windows/RNSVG/Fabric/RenderableView.cpp

```diff
@@ -0,0 +1,174 @@
+#include "pch.h"
+#include "RenderableView.h"
+
+#include "SvgView.h"
+#include "D2DHelpers.h"
+
+namespace winrt::Microsoft::ReactNative {
+
+void WriteValue(IJSValueWriter const &writer, const D2D1_SVG_LENGTH &value) noexcept {
+  if (value.units == D2D1_SVG_LENGTH_UNITS::D2D1_SVG_LENGTH_UNITS_PERCENTAGE) {
+    writer.WriteString(winrt::to_hstring(std::to_string(value.value) + "%"));
+  } else {
+    writer.WriteDouble(value.value);
+  }
+}
+
+void ReadValue(IJSValueReader const &reader, /*out*/ D2D1_SVG_LENGTH &value) noexcept {
+  switch (reader.ValueType()) {
+    case JSValueType::String: {
+      auto str = to_string(reader.GetString());
+
+      auto strLength{str.size()};
+      if (strLength == 0 || str == "normal") {
+        value.units = D2D1_SVG_LENGTH_UNITS::D2D1_SVG_LENGTH_UNITS_NUMBER;
+        value.value = 0.0;
+        return;
+      } else if (str.back() == '%') {
+        value.units = D2D1_SVG_LENGTH_UNITS::D2D1_SVG_LENGTH_UNITS_PERCENTAGE;
+        value.value = std::stof(str.substr(0, strLength - 1), nullptr);
+        return;
+      } else if (strLength > 2) {
+        auto end{strLength - 2};
+        auto lastTwo{str.substr(end)};
+
+        if (lastTwo == "px") {
+          value.units = D2D1_SVG_LENGTH_UNITS::D2D1_SVG_LENGTH_UNITS_NUMBER;
+        } else if (lastTwo == "em") {
+          value.units = D2D1_SVG_LENGTH_UNITS::D2D1_SVG_LENGTH_UNITS_NUMBER;
+          // value.unit = D2D1_SVG_LENGTH_UNITS::D2D1_SVG_LENGTH_UNITS_EM;
```

Comment:
did you mean to leave these in?

Reply:
Added a comment at the top of the block.